amcl/src/amcl_node.cpp

```diff
@@ -1687,41 +1660,38 @@ AmclNode::handleInitialPosesMessage(const move_base_msgs::PoseWithCovarianceStam
 
   initial_poses_hyp_.erase(initial_poses_hyp_.begin(), initial_poses_hyp_.end());
 
-  for(auto pose : msg.poses)
+  tf::Pose pose_old, pose_new;
+  tf::poseMsgToTF(msg.pose.pose, pose_old);
+  pose_new = pose_old * tx_odom;
+
+  stream << " ";
+  stream << pose_new.getOrigin().x() << " ";
+  stream << pose_new.getOrigin().y() << " ";
+  stream << getYaw(pose_new) << ",";
+
+  // Re-initialize the filter
+  pf_vector_t pf_init_pose_mean = pf_vector_zero();
+  pf_init_pose_mean.v[0] = pose_new.getOrigin().x();
+  pf_init_pose_mean.v[1] = pose_new.getOrigin().y();
+  pf_init_pose_mean.v[2] = getYaw(pose_new);
+  pf_matrix_t pf_init_pose_cov = pf_matrix_zero();
+
+  // Copy in the covariance, converting from 6-D to 3-D
+  for(int i=0; i<2; i++)
   {
-	  tf::Pose pose_old, pose_new;
-	  tf::poseMsgToTF(pose.pose, pose_old);
-	  pose_new = pose_old * tx_odom;
-
-	  stream << " ";
-	  stream << pose_new.getOrigin().x() << " ";
-	  stream << pose_new.getOrigin().y() << " ";
-	  stream << getYaw(pose_new) << ",";
-
-	  // Re-initialize the filter
-	  pf_vector_t pf_init_pose_mean = pf_vector_zero();
-	  pf_init_pose_mean.v[0] = pose_new.getOrigin().x();
-	  pf_init_pose_mean.v[1] = pose_new.getOrigin().y();
-	  pf_init_pose_mean.v[2] = getYaw(pose_new);
-	  pf_matrix_t pf_init_pose_cov = pf_matrix_zero();
-
-	  // Copy in the covariance, converting from 6-D to 3-D
-	  for(int i=0; i<2; i++)
-	  {
-	    for(int j=0; j<2; j++)
-	    {
-	      pf_init_pose_cov.m[i][j] = pose.covariance[6*i+j];
-	    }
-	  }
-	  pf_init_pose_cov.m[2][2] = pose.covariance[6*5+5];
-
-	  amcl_hyp_t hyp;
-
-	  hyp.pf_pose_mean = pf_init_pose_mean;
-	  hyp.pf_pose_cov = pf_init_pose_cov;
-
-	  initial_poses_hyp_.push_back(hyp);
+    for(int j=0; j<2; j++)
+    {
+      pf_init_pose_cov.m[i][j] = msg.pose.covariance[6*i+j];
+    }
   }
+  pf_init_pose_cov.m[2][2] = msg.pose.covariance[6*5+5];
+
+  amcl_hyp_t hyp;
+
+  hyp.pf_pose_mean = pf_init_pose_mean;
+  hyp.pf_pose_cov = pf_init_pose_cov;
+
+  initial_poses_hyp_.push_back(hyp);
```

Comment:
Do we still need to maintain initial_poses_hyp_ vector? I think we'll always have one initial pose hyp to maintain.

Reply:
We don't need a vector but will need a way to keep track of initial_pose_hyp. I can make the change and remove it.